types/src/index.ts

```diff
@@ -33,45 +158,46 @@ function printDefinitions(root: StructureGroups): string {
   let source = printNodeList(nodes);
   sources.set(sourcePath, source);
 
-  // Build TypeScript program from source files and overrides. Importantly,
-  // these are in the same program, so we can use nodes from one in the other.
-  let program = createMemoryProgram(sources);
-  let checker = program.getTypeChecker();
-  let sourceFile = program.getSourceFile(sourcePath);
-  assert(sourceFile !== undefined);
-
   // Run post-processing transforms on program
-  let result = ts.transform(sourceFile, [
+  source = transform(sources, sourcePath, (program, checker) => [
     // Run iterator transformer before overrides so iterator-like interfaces are
     // still removed if they're replaced in overrides
     createIteratorTransformer(checker),
     createOverrideDefineTransformer(program, replacements),
   ]);
-  assert.strictEqual(result.transformed.length, 1);
 
   // We need the type checker to respect our updated definitions after applying
   // overrides (e.g. to find the correct nodes when traversing heritage), so
   // rebuild the program to re-run type checking.
   // TODO: is there a way to re-run the type checker on an existing program?
-  source = printer.printFile(result.transformed[0]);
-  program = createMemoryProgram(new Map([[sourcePath, source]]));
-  checker = program.getTypeChecker();
-  sourceFile = program.getSourceFile(sourcePath);
-  assert(sourceFile !== undefined);
+  source = transform(
+    new Map([[sourcePath, source]]),
+    sourcePath,
+    (program, checker) => [
+      // Run global scope transformer after overrides so members added in
+      // overrides are extracted
+      createGlobalScopeTransformer(checker),
+      createCommentsTransformer(standards),
+      createAmbientTransformer(),
+      // TODO(polish): maybe flatten union types?
+    ]
+  );
 
-  result = ts.transform(sourceFile, [
-    // Run global scope transformer after overrides so members added in
-    // overrides are extracted
-    createGlobalScopeTransformer(checker),
-    // TODO(polish): maybe flatten union types?
-  ]);
-  assert.strictEqual(result.transformed.length, 1);
+  checkDiagnostics(new Map([[sourcePath, source]]));
+
+  const exportable = transform(
+    new Map([[sourcePath, source]]),
+    sourcePath,
+    () => [createExportableTransformer()]
+  );
```

Comment:
I just wanted to make sure both ambient and exportable declarations were in the right format, regardless of changes in the source format (overrides being part of that). It's not much overhead, so I think keeping both makes sense, in case we need to make changes in the future (removing ambient declarations in favour of DOM interop/removing exportable declarations/or anything else like that)